client/components/ConversationSearch/index.js

```diff
@@ -1,13 +1,43 @@
 import React from 'react';
+import { useDispatch } from 'react-redux'
+import { searchChatsRequest, emptySearchChatsRequest } from '../../store/actions/search-chats';
 import './ConversationSearch.css';
 
-export default function ConversationSearch() {
+export default function ConversationSearch({ setSearchedChats, searchedChats }) {
+
+  const dispatch = useDispatch();
+    
+  const getChat = async (search) => {
+    if (search !== '') {
+      await dispatch(searchChatsRequest(search))
+    } else {
+      setSearchedChats([]);
+    }
+  }
+
+  const handleSearch = async (e) => {
+    if (e.target.value === '') {
+      await dispatch(emptySearchChatsRequest())
+      // setSearchedChats(null);
+    }
+  };
+
+  React.useEffect(() => {
+    window.addEventListener('search', handleSearch);
```

Comment:
Omo! I think this is expensive for the DOM
I think you might want to use React controlled form for this.

Reply:
Removed